server/src/test/java/org/opensearch/cluster/coordination/JoinTaskExecutorTests.java

```diff
@@ -582,6 +582,14 @@ public void testPreventJoinClusterWithRemoteStoreNodeWithPartialAttributesJoinin
             );
             assertTrue(
                 e.getMessage().equals("joining node [" + joiningNode + "] doesn't have the node attribute [" + nodeAttribute.getKey() + "]")
+                    || e.getMessage()
+                        .equals(
+                            "a remote store node ["
+                                + joiningNode
+                                + "] is trying to join a remote store cluster with incompatible node attributes in comparison with existing node ["
+                                + currentState.getNodes().getNodes().values().stream().findFirst().get()
+                                + "]"
+                        )
```

Comment:
are we planning to add some more tests? To cover the new combinations of configuration.

Reply:
Yes, lets add test to cover the new combination where remote state is enabled without remote store for segments

Reply:
Added tests